drivers/power/ltc7841/ltc7841.h

```diff
@@ -0,0 +1,189 @@
+/***************************************************************************//**
+ *   @file   ltc7841.h
+ *   @brief  Header file of LTC7841 Driver.
+ *   @author Marvin Cabuenas (marvinneil.cabuenas@analog.com)
+********************************************************************************
+ * Copyright 2024(c) Analog Devices, Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES, INC. “AS IS” AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO
+ * EVENT SHALL ANALOG DEVICES, INC. BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+ * OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __LTC7841_H__
+#define __LTC7841_H__
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include "no_os_i2c.h"
+#include "no_os_irq.h"
+#include "no_os_print_log.h"
+#include "no_os_delay.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+#include "no_os_units.h"
+#include "no_os_alloc.h"
+
+/** X is set based on the pull configuration of the ADDR pin */
+#define TWO_BYTES_LENGTH            2
+#define LTC7841_ADDR                (0x20)
+#define REPEATED_START_MODE         1
+#define I2C_WRITE_LENGTH_FOR_READ   1
+#define I2C_DEFAULT_ARRAY_SIZE      2
+#define STOP_MODE                   0
+#define LTC7841_DEFAULT_ADDRESS_0   0x5A
+#define LTC7841_DEFAULT_ADDRESS_1   0x5B
+#define LTC7841_TOTAL_REGISTERS     0x16
+#define LTC7841_BYTE_LENGTH         1
+#define LTC7841_WORD_LENGTH         2
+#define LTC7841_I2C_WR_FRAME_SIZE   3
+/* this denotes the rsense scale of LTC7841 2(both channels) * 50 uV/ bit / 0.002 (rsense resistor), unit in uA*/
+#define CURRENT_SCALE               (20)
```

Comment:
Since RSENSE will now be a param value, you can compute this instead.

Reply:
Also it's not used in the driver but in the iio_driver, maybe just add this as the scale attribute in the iio_driver.